src/assets/AssetsContractController.ts

```diff
@@ -128,32 +116,76 @@ export class AssetsContractController extends BaseController<
    * @param index - A collectible counter less than `balanceOf(selectedAddress)`.
    * @returns Promise resolving to token identifier for the 'index'th asset assigned to 'selectedAddress'.
    */
-  getCollectibleTokenId(
+  getERC721CollectibleTokenId(
     address: string,
     selectedAddress: string,
     index: number,
   ): Promise<string> {
-    const contract = this.web3.eth.contract(abiERC721).at(address);
     return this.erc721Standard.getCollectibleTokenId(
-      contract,
+      address,
       selectedAddress,
       index,
     );
   }
 
+  async getTokenStandardAndDetails(
+    address: string,
+    userAddress: string,
+    tokenId?: string,
+  ) {
```

Comment:
Is it possible to provide a return type for this? I think that would make this easier to understand.

Reply:
done here: https://github.com/MetaMask/controllers/pull/667/commits/8effcafddb6ca32d9566050ba50db569ceb00f07